lib/src/view/relation/following_screen.dart

```diff
@@ -0,0 +1,372 @@
+    final following = ref.watch(followingProvider);
+    final relationState = ref.watch(relationCtrlProvider);
+    final followingOnlines = relationState.when(
+      data: (data) => data.followingOnlines,
+      error: (error, stackTrace) {
+        debugPrint(
+          'SEVERE: [FollowingScreen] could not load following online users; $error\n$stackTrace',
+        );
+        return List<LightUser>.empty().toIList();
+      },
+      loading: () {
+        return List<LightUser>.empty().toIList();
+      },
+    );
+
+    return following.when(
+      data: (data) {
+        IList<User> followingUsers = data;
+        return StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            if (followingUsers.isEmpty) {
+              return const Center(
+                child: Text("You are not following any user"),
+              );
+            }
+            return SafeArea(
+              child: ListView(
+                children: [
+                  ListSection(
+                    hasLeading: true,
+                    children: [
+                      for (final user in followingUsers)
+                        Slidable(
+                          endActionPane: ActionPane(
+                            motion: const ScrollMotion(),
+                            extentRatio: 0.3,
+                            children: [
+                              SlidableAction(
+                                onPressed: (BuildContext context) {
+                                  showAdaptiveDialog<void>(
+                                    context: context,
+                                    builder: (context) => _UnfollowDialog(
+                                      title: Text(
+                                        'Unfollow ${user.username}?',
+                                      ),
+                                      onAccept: () async {
+                                        final oldState = followingUsers;
+                                        setState(() {
+                                          followingUsers =
+                                              followingUsers.removeWhere(
+                                            (v) => v.id == user.id,
+                                          );
+                                        });
+
+                                        final res = await ref
+                                            .read(relationRepositoryProvider)
+                                            .unfollow(user.username);
+                                        if (res.isError) {
+                                          setState(() {
+                                            followingUsers = oldState;
+                                          });
+                                        } else {
+                                          ref
+                                              .read(
```

Comment:
Same as other comment, the whole caching logic seems too complex here. Normally just invalidating the `followingProvider` should be enough to trigger a screen refresh.

Perhaps you could take the account preferences notifier as an example: https://github.com/lichess-org/mobile/blob/38fa6bc675f0c580eeced938ff7166e07a101ed1/lib/src/model/account/account_preferences.dart#L55

It works the same way: the initial data is fetched async from the server, and methods that update the state trigger an `invalidateSelf` to force fetching new data: https://github.com/lichess-org/mobile/blob/38fa6bc675f0c580eeced938ff7166e07a101ed1/lib/src/model/account/account_preferences.dart#L90

Reply:
Yes, I understand.
But if we add await `Future.delayed(const Duration(seconds: 1));` before line 89 in the account_preferences, simulating a slow POST request, the delay will affect the toggle component, causing it to wait for 1 second before switching. The same behavior would occur when removing a user tile on the following screen.

Reply:
Yes it's not perfect, I agree, but in the vast majority of cases it will work smoothly. 

UX can be improved I guess: set an opacity, or any other way to indicate something is loading. It will have the big advantage to be simple, compared to complex caching.